wolfcrypt/src/rsa.c

```diff
@@ -1157,7 +1157,7 @@ static int RsaPad_PSS(const byte* input, word32 inputLen, byte* pkcsBlock,
         return PSS_SALTLEN_E;
     }
 #endif
-    if ((int)pkcsBlockLen - hLen < saltLen + 2) {
+    if ((int)pkcsBlockLen - (int)inputLen - hLen < saltLen + 2) {
```

Comment:
Line 1151: saltLen = pkcsBlockLen - hlen - 2;
This won't work!

Need another check.